brave/brave-core
# Added on-demand mode for speedreader.

Resolves https://github.com/brave/brave-browser/issues/29570

## Submitter Checklist:

  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`

## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on

## Test Plan:

1. Enable Distillability Service be passing ```--enable-distillability-service``` in the command line or by enabling ```ReaderMode``` feature
2. Check news sites which not automatically speedreaded (reader mode button should appear)
    Some urls that are suitable are listed in https://github.com/brave/brave-browser/issues/23418
    * There is no guarantee that every page that shows the "reading mode" button can be distilled.
